presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -526,6 +527,69 @@ private static SqlScalarFunction decimalRoundNFunction()
                 .build();
     }
 
+    private static SqlScalarFunction decimalTruncateFunction()
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name("truncate")
+                .literalParameters("num_precision", "num_scale", "result_precision")
+                .longVariableConstraints(
+                        longVariableCalculation("result_precision", "min(38, num_precision)"))
+                .argumentTypes("decimal(num_precision, num_scale)", BIGINT)
+                .returnType("decimal(result_precision, num_scale)")
```

Comment:
can the precision be bigger than original? 

can't we do a better job with the result scale?